zcash_client_backend/src/data_api.rs

```diff
@@ -66,72 +92,106 @@ pub trait WalletRead {
             .map(|oo| oo.flatten())
     }
 
+    /// Returns the block height in which the specified transaction was mined.
     fn get_tx_height(&self, txid: TxId) -> Result<Option<BlockHeight>, Self::Error>;
 
-    fn get_address<P: consensus::Parameters>(
-        &self,
-        params: &P,
-        account: AccountId,
-    ) -> Result<Option<PaymentAddress>, Self::Error>;
+    /// Returns the payment address for the specified account, if the account
+    /// identifier specified refers to a valid account for this wallet.
+    fn get_address(&self, account: AccountId) -> Result<Option<PaymentAddress>, Self::Error>;
 
-    fn get_extended_full_viewing_keys<P: consensus::Parameters>(
+    /// Returns all extended full viewing keys known about by this wallet
+    fn get_extended_full_viewing_keys(
         &self,
-        params: &P,
-    ) -> Result<Vec<ExtendedFullViewingKey>, Self::Error>;
+    ) -> Result<HashMap<AccountId, ExtendedFullViewingKey>, Self::Error>;
 
-    fn is_valid_account_extfvk<P: consensus::Parameters>(
+    /// Checks whether the specified extended full viewing key is
+    /// associated with the account.
+    fn is_valid_account_extfvk(
         &self,
-        params: &P,
         account: AccountId,
         extfvk: &ExtendedFullViewingKey,
     ) -> Result<bool, Self::Error>;
 
-    fn get_balance(&self, account: AccountId) -> Result<Amount, Self::Error>;
+    /// Returns the wallet balance for the specified account.
+    ///
+    /// This balance amount is the raw balance of all transactions in known
+    /// mined blocks, irrespective of confirmation depth. A default implementation
+    /// is provided, but implementations of this trait may override the default
+    /// implementation for efficiency.
+    fn get_balance(&self, account: AccountId) -> Result<Amount, Self::Error> {
+        if let Some((_, tip)) = self.block_height_extrema()? {
+            self.get_verified_balance(account, tip)
+        } else {
+            Ok(Amount::zero())
+        }
+    }
```

Comment:
This is incorrect. `get_verified_balance` includes the "only count transactions mined at least 10 blocks ago" - or at least, that is the intention of the word `verified`.

Reply:
I think we should get rid of `get_verified_balance` entirely and jus have `get_balance` take a block height.

Reply:
From the perspective of the DAA accessing wallet data, I agree that we only need a `get_balance_at(height)` API; then the business logic can deal with the definition of "verified".